src/main/java/modal/StringAddCalculator.java

```diff
@@ -0,0 +1,59 @@
+package modal;
+
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    private static String DELIMITER = "[,:]";
+    private static final String CUSTOM_DELIMITER_REGEX = "//(.)\\n(.*)";
+
+    public static int splitAndSum(String inputString) {
+        if(checkNullinput(inputString)) {
+            return 0;
+        }
+        String[] splitString = makeSplitString(inputString);
+        return sumNumbers(splitString);
+    }
+
+    private static int checkMinus(String stringNum) {
+        int number = 0;
+        try {
+            number = Integer.parseInt(stringNum);
+            if(number < 0) {
+                throw new RuntimeException();
+            }
+        } catch (NumberFormatException e) {
+            throw new NumberFormatException();
+        }
+
+        return number;
+    }
+
+    private static int sumNumbers(String[] splitString) {
+        int sum = 0;
+        for( String num :  splitString) {
+            sum += checkMinus(num);
+        }
+
+        return sum;
+    }
+
+    private static String[] makeSplitString(String inputString) {
+        String[] splitString = null;
+        Matcher m = Pattern.compile(CUSTOM_DELIMITER_REGEX).matcher(inputString);
```

Comment:
Pattern.compile() 내부 구현을 보면 
```
// Pattern
public static Pattern compile(String regex) {
    return new Pattern(regex, 0);
}
```
makeSplitString() 메서드를 호출할때마다 String을 패턴 매칭을 위해
매번 성능이 비싼 Pattern 인스턴스를 생성하고 Pattern 인스턴스는 한번 쓰고 버려지기 때문에 성능적으로 좋지 않습니다.  뿐더러 자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있습니다.

If a pattern is to be used multiple times, compiling it once and reusing
it will be more efficient than invoking this method each time

매번 불필요한 객체 생성을 피하도록 개선하면 좋을 것 같아요 :)

Reply:
적용 완료 하였습니다.